API.md

```diff
@@ -0,0 +1,387 @@
+
+  - `regular`, `bold`, `italic` & `bolditalic` should point to the font file matching their respective font style. At least one of them MUST be specified.
+
+  - `size` sets the rendering size, in points. Defaults to 12pt if unset. Can be a decimal value.
+
+  - `px` sets the rendering size, in pixels. Optional. Takes precedence over size if specified.
+
+  - `top`, `bottom`, `left` & `right` set the margins used to define the display area. Defaults to 0, i.e., the full screen, starting at the top-left corner.
+
+    NOTE: If a negative value is supplied, counts backward from the opposite edge. Mostly useful with top & left to position stuff relative to the bottom right corner.
+
+  - `padding` can optionally be set to ensure the drawing area on both sides of the printed content is padded with the background color on one or both axis.
+
+    Available padding axis: `HORIZONTAL`, `VERTICAL`, or `BOTH` (Defaults to `NONE`). Useful to avoid overlaps on consecutive prints at the same coordinates.
+
+  - If style is specified, it dictates the default font style to use (e.g., `REGULAR`, `BOLD`, `ITALIC` or `BOLD_ITALIC`). Defaults to `REGULAR`.
+
+  - If `format` is specified, instead of the default style, the underscore/star Markdown syntax will be honored to set the font style (i.e., `*italic*`, `**bold**` & `***bold italic***`).
+
+  - If `notrunc` is specified, truncation will be considered a failure.
+
+    NOTE: This may not prevent drawing/refreshing the screen if the truncation couldn't be predicted at compute time!
+
+    On the CLI, this will prevent you from making use of the returned computation info, as this will chain a CLI abort.
+
+  - If `compute` is specified, no rendering will be done, and only the line-breaking computation pass will run. You'll generally want to use that combined with `-l`, `--linecount`.
+
+  Honors `-h`, `--invert`; `-f`, `--flash`; `-c`, `--clear`; `-W`, `--waveform`; `-D`, `--dither`; `-H`, `--nightmode`; `-b`, `--norefresh`; `-m`, `--centered`; `-M`, `--halfway`; `-o`, `--overlay`; `-T`, `--fgless`; `-O`, `--bgless`; `-C`, `--color`; `-B`, `--background`; `-l`, `--linecount`.
+
+  Examples:
+
+  * If compiled with `FBINK_FOR_KINDLE`:
+
+    ```sh
+    fbink -t regular=/usr/java/lib/fonts/Caecilia_LT_65_Medium.ttf,bold=/usr/java/lib/fonts/Caecilia_LT_75_Bold.ttf,size=24,top=100,bottom=500,left=25,right=50,format "Hello **world**!"
+    ```
+
+    Will use Caecilia to print "Hello world!" at 24pt in a display area starting from 100px down the top of the screen to 500px before the bottom of the screen, from 25px of the left edge of the screen until 50px before the right edge. Honoring the Markdown syntax, "Hello" will be printed with the Regular font style, while "world" will use the Bold font style. You will NOT be able to use obfuscated or encrypted fonts.
+
+  * If not compiled with `FBINK_FOR_KINDLE`:
```

Comment:
For humans, that actually means a Kobo build ;).

This is possibly applicable to other `FBINK_FOR` mentions in here: make it human readable an name the target platform explicitly instead of trying to grok the convoluted preprocessing rules for those ;p

Reply:
So, basically we have "for Linux" "for Kindle" and "for Kobo", which I can assume are mutually exclusive?

Reply:
Kindle/Kobo/Cervantes/reMarkable/PocketBook

And Linux for "not an e-Ink device" ;).

And, yeah, they're all mutually exclusive (the default being Kobo, not that it should matter, but it may help you follow the preprocessing ladders).

Reply:
Got it. Fixed in 7e4ab88586c01abc352eac313b266afb759e56e1